src/synthesizer/grid.py

```diff
@@ -337,20 +348,84 @@ def _get_lines_grid(self, read_lines):
                 self.available_lines = flatten_linelist(read_lines)
             else:
                 # If not, read all available lines
-                self.available_lines, _ = self.get_grid_line_ids()
+                self.available_lines, _ = self._get_line_ids_from_file()
+
+            # Create an entry for each spectra type
+            for spectra in self.available_spectra:
+                self.line_lums[spectra] = {}
+                self.line_conts[spectra] = {}
+
+            # We read the lines themselves into "nebular" so make sure this
+            # exists
+            if "nebular" not in self.line_lums:
+                raise exceptions.GridError(
+                    "No nebular spectra found. The nebular spectra is"
+                    " required for storing lines. Either you have explictly"
+                    " requested a subset of spectra without nebular, or "
+                    "something is wrong with the grid file."
+                )
+
+            # Read the line wavelengths
+            lines_outside_lam = []
+            for line in self.available_lines:
+                self.line_lams[line] = hf["lines"][line].attrs["wavelength"]
+
+                # Ensure this wavelength is within the wavelength array of the
+                # grid
+                if (
+                    self.line_lams[line] < self.lam[0]
+                    or self.line_lams[line] > self.lam[-1]
+                ):
+                    lines_outside_lam.append(line)
+
+            # If we have lines outside the wavelength range of the grid
+            # warn the user
+            if len(lines_outside_lam) > 0:
+                warn(
+                    "The following lines are outside the wavelength "
+                    f"range of the grid: {lines_outside_lam}"
+                )
 
-            # Read in the lines
+            # Read the lines into the nebular entry
             for line in self.available_lines:
-                self.lines[line] = {}
-                self.lines[line]["wavelength"] = hf["lines"][line].attrs[
-                    "wavelength"
-                ]
-                self.lines[line]["luminosity"] = hf["lines"][line][
+                self.line_lums["nebular"][line] = hf["lines"][line][
                     "luminosity"
                 ][:]
-                self.lines[line]["continuum"] = hf["lines"][line]["continuum"][
-                    :
-                ]
+                self.line_conts["nebular"][line] = hf["lines"][line][
```

Comment:
I think this should be `nebular_continuum`. `continuum` includes both the nebular and transmitted continuum.

Reply:
You're going to need to expand on what you mean here. 

My understanding: The `"nebular_continuum"` key by definition is `line_conts["nebular"]`. The `"nebular"` key contains the lines (but these also need to be included in "`line_cont"` since the `"nebular"` key is ignored when `fesc_ly_alpha` is `<1.0` (because the nebular becomes a combination not an extraction).

Should we be evaluating all continua from their respective spectra instead of reading them? That's nice and simple if so.

Reply:
The `continuum` in the grid file is the sum of the transmitted continuum and the nebular continuum. Since here, you're trying to populate just the nebular component you need to use `nebular_continuum` instead. 

If it's simpler you could evaluate the continuum from the spectra except that that doesn't work for nebular since the spectra contains the line and continuum.

Reply:
I've now modified this so that it evaluates the spectra. With the caveats that:
- `"nebular"` extracts the continuum from `"nebular_continuum"`.
- `"linecont"` has a continuum of 0.